datasets/datasets_decorator.py

```diff
@@ -0,0 +1,23 @@
+import functools
+from typing import Optional
+
+from datasets.context import Context
+
+from .dataset_plugin import DatasetPlugin
+
+
+def dataset(context: Optional[Context] = None, **dataset_kwargs):
+    def step_decorator(func):
+        @functools.wraps(func)
+        def step_wrapper(*args, **kwargs):
+            self = args[0]
+            dataset = DatasetPlugin.from_keys(context=context, **dataset_kwargs)
+
+            if not hasattr(self, dataset._attribute_name):
```

Comment:
perhaps handled elsewhere, but should we throw a warning or something if `hasattr(self, dataset._attribute_name) == True`? this would be an attempt to overwrite the objects dataset attribute, in this case we would stick with the first applied

Reply:
This is a good catch.

**Options:**
1. Not allow dataset attribute overwrite and throw `InvalidOperationException`
2. Allow dataset attribute overwrites.
The current behavior silently ignores an overwrite, which I think is wrong and confusing.

I'm favoring option 2 for more expressibility.  What do you think @busheycj ?

Reply:
I agree with option 2